src/sync.c

```diff
@@ -115,6 +116,13 @@ mpd_sync_send_command_v(struct mpd_async *async, const struct timeval *tv0,
 		success = mpd_async_send_command_v(async, command, copy);
 		va_end(copy);
 
+		/* no characters were written to async buffer */
+		if ((mpd_async_events(async) & MPD_ASYNC_EVENT_WRITE) == 0) {
```

Comment:
This is a bad check for the error.
1. the buffer could already have some data but not enough for the new command, thus making this a false positive
2. the command could fit, but it failed because there was a previous error, thus overwriting the existing error condition
And it doesn't check the `success` flag.

Reply:
Hello Max Kellermann, thanks for the review.
1. Maybe i'm not understanding correctly, but if the buffer already have some data to write, then `mpd_async_events()` will have the `MPD_ASYNC_EVENT_WRITE` flag set; that is why i did that test. In other words, that test is intended to see if more space will be made available in the buffer by calling `mpd_sync_io()`.
I have updated the comment on line 119 to reflect this.

2. This is already done at `mpd_async_set_error()`; if an error is already set, it returns false and doesn't change it (mentioned in the declaration of this function in src/iasync.h). If you want another name for this function to reflect this, let me know.

The success flag was an oversight; i've corrected it with a new commit.

new commit is https://github.com/MusicPlayerDaemon/libmpdclient/pull/54/commits/8f46a3e4c9453658a5cfe0a177d5f6c88583b346